projects/ui/src/util/FormTxns.ts

```diff
@@ -0,0 +1,587 @@
+          ]
+        );
+
+        callData.push(encoded);
+        steps.push(async (_amountInStep, _context) => ({
+          name: 'enrootDeposit',
+          amountOut: _amountInStep,
+          prepare: () => ({
+            target: beanstalk.address,
+            callData: encoded,
+          }),
+          decode: (data: string) =>
+            beanstalk.interface.decodeFunctionData('enrootDeposit', data),
+          decodeResult: (result: string) =>
+            beanstalk.interface.decodeFunctionResult('enrootDeposit', result),
+        }));
+      } else if (crates?.length > 1) {
+        const encoded = beanstalk.interface.encodeFunctionData(
+          'enrootDeposits',
+          [
+            urToken.address,
+            crates.map((crate) => crate.season.toString()),
+            crates.map((crate) =>
+              urToken.fromHuman(crate.amount.toString()).toBlockchain()
+            ),
+          ]
+        );
+
+        callData.push(encoded);
+        steps.push(async (amountInStep, _context) => ({
+          name: 'enrootDeposits',
+          amountOut: amountInStep,
+          prepare: () => ({
+            target: beanstalk.address,
+            callData: encoded,
+          }),
+          decode: (data: string) =>
+            beanstalk.interface.decodeFunctionData('enrootDeposits', data),
+          decodeResult: (result: string) =>
+            beanstalk.interface.decodeFunctionResult('enrootDeposits', result),
+        }));
+      }
+    });
+
+    return {
+      getSteps: () => steps,
+      estimateGas: () => beanstalk.estimateGas.farm([...callData]),
+    };
+  };
+
+  static transferToken = (
+    sdk: BeanstalkSDK,
+    account: string,
+    token: Token,
+    amount: TokenValue,
+    from: FarmFromMode,
+    to: FarmToMode
+  ): {
+    getSteps: () => StepGenerator[];
+    estimateGas: () => Promise<ethers.BigNumber>;
+  } => {
+    const step = new sdk.farm.actions.TransferToken(
+      token.address,
+      account,
+      from,
+      to
+    );
+
+    const estimateGas = () =>
+      sdk.contracts.beanstalk.estimateGas.transferToken(
+        token.address,
+        account,
+        amount.toBigNumber(),
+        from,
+        to
+      );
+
+    return {
+      getSteps: () => [step],
+      estimateGas,
+    };
+  };
+
+  private static implied: { [key in FormTxn]: FormTxn[] | undefined } = {
+    [FormTxn.ENROOT]: [FormTxn.MOW],
+    [FormTxn.PLANT]: [FormTxn.MOW],
+    [FormTxn.MOW]: undefined,
+    [FormTxn.CLAIM]: undefined,
+    [FormTxn.RINSE]: undefined,
+    [FormTxn.HARVEST]: undefined,
+  };
+
+  private static functionMap: { [key in FormTxn]: FormTxnFunction<key> } = {
+    [FormTxn.MOW]: FormTxnBuilder.mow,
+    [FormTxn.PLANT]: FormTxnBuilder.plant,
+    [FormTxn.ENROOT]: FormTxnBuilder.enroot,
+    [FormTxn.RINSE]: FormTxnBuilder.rinse,
+    [FormTxn.HARVEST]: FormTxnBuilder.harvest,
+    [FormTxn.CLAIM]: FormTxnBuilder.claim,
+  };
```

Comment:
Okay, after reviewing how this is used. Here is an updated suggestion.

This can be pulled out into a factory into it's own file.

Ex, 

```javascript
// Some strategy folder 
class HarvestStepStrategy {
  constructor(sdk, params) {
    ...
  }
  getSteps() {
    ...
  }
  estimateGas() {
    ...
  }
}

class RinseStepStrategy {
  constructor(sdk, params) {
    ...
  }
  getSteps() {
    ...
  }
  estimateGas() {
    ...
  }
}

class ClaimStepStrategy {
  constructor(sdk, params) {
    ...
  }
  getSteps() {
    ...
  }
  estimateGas() {
    ...
  }
}

class MowStepStrategy {
  constructor(sdk, params) {
    ...
  }
  getSteps() {
    ...
  }
  estimateGas() {
    ...
  }
}

class PlantStepStrategy {
  constructor(sdk, params) {
    ...
  }
  getSteps() {
    ...
  }
  estimateGas() {
    ...
  }
}


class EnrootStepStrategy {
  constructor(sdk, params) {
    ...
  }
  getSteps() {
    ...
  }
  estimateGas() {
    ...
  }
}

class FormTxnStepFactory {
  getStep(step) {
    switch(step) {
      case FormTxn.MOW:{ 
        return MowStepStrategy
      }
      case FormTxn.PLANT:{ 
        return PlantStepStrategy
      }
      case FormTxn.ENROOT:{ 
        return EnrootStepStrategy
      }
      case FormTxn.RINSE:{ 
        return RinseStepStrategy
      }
      case FormTxn.HARVEST:{ 
        return HarvestStepStrategy
      }
      case FormTxn.CLAIM:{ 
        return ClaimStepStrategy
      }
      default:
        throw new Error('Step does not exist')
    }
  }
}
```

If you have the above code, you can remove the following code: 
* https://github.com/BeanstalkFarms/Beanstalk/blob/179fa41fa2123ee954718c52d62d843c96e6a597/projects/ui/src/util/FormTxns.ts#L369-L376
* https://github.com/BeanstalkFarms/Beanstalk/blob/179fa41fa2123ee954718c52d62d843c96e6a597/projects/ui/src/util/FormTxns.ts#L385-L387
* https://github.com/BeanstalkFarms/Beanstalk/blob/179fa41fa2123ee954718c52d62d843c96e6a597/projects/ui/src/util/FormTxns.ts#L117-L358

Then you can refactor all calls to `FormTxnBuilder.getFunction`. 

Here is an example of how it can be refactored:
https://github.com/BeanstalkFarms/Beanstalk/blob/179fa41fa2123ee954718c52d62d843c96e6a597/projects/ui/src/components/Barn/Actions/Rinse.tsx#L214-L219

```javascript
// Call Exmaple
const params = {
  tokenIds: farmerBarn.balances.map((bal) => bal.token.id.toString()),
  toMode: values.destination,
};
const stepKlass = FormTxnStepFactory.getStep(FormTxn.RINSE);
const rinse = new stepKlass(sdk, params);
const rinseStep = rinse.getSteps();
```